src/main/scala/com/gu/newsletterlistcleanse/braze/BrazeModels.scala

```diff
@@ -0,0 +1,100 @@
+  implicit val brazeResponseDecoder: Decoder[BrazeResponse] = deriveDecoder
+}
+
+case class BrazeError(code: Int, body: String)
+
+object BrazeError {
+  def apply(response: HttpResponse[String]): BrazeError = BrazeError(response.code, response.body)
+}
+
+case class BrazeNewsletterSubscriptionsUpdate(externalId: String,
+                                              newsletterSubscriptions: Map[EmailNewsletter, Boolean])
+
+object BrazeNewsletterSubscriptionsUpdate {
+  implicit val subscriptionUpdateEncoder: Encoder[BrazeNewsletterSubscriptionsUpdate] =
+    new Encoder[BrazeNewsletterSubscriptionsUpdate] {
+      override def apply(update: BrazeNewsletterSubscriptionsUpdate): Json = {
+        val jsonSubs = update.newsletterSubscriptions.map {
+          // This case only relevant whilst migrating email_subscribe_today_uk custom attribute
+          case (newsLetter: EmailNewsletter, isSubscribed) if newsLetter == EmailNewsletters.guardianTodayUk =>
+            Map(
+              newsLetter.brazeSubscribeAttributeName -> Json.fromBoolean(isSubscribed),
+              "email_subscribe_today_uk" -> Json.fromBoolean(isSubscribed)
+            )
+          case (newsLetter, isSubscribed) =>
+            Map(newsLetter.brazeSubscribeAttributeName -> Json.fromBoolean(isSubscribed))
+        }.fold(Map.empty)(_ ++ _)
+        (jsonSubs ++ Map("external_id" -> Json.fromString(update.externalId))).asJson
+      }
+  }
+}
+
+case class BrazeEventProperties(campaign_name: String)
+
+case class BrazeEvent(external_id: String,
+                      name: String,
+                      time: String,
+                      properties: BrazeEventProperties,
+                      updateExistingOnlyField: Boolean = false)
+
+object BrazeEvent {
+  implicit val brazeEventEncoder: Encoder[BrazeEvent] = new Encoder[BrazeEvent] {
+    override def apply(event: BrazeEvent): Json = Json.obj(
+        "external_id" -> Json.fromString(event.external_id),
+        "name" -> Json.fromString(event.name),
+        "properties" -> Map("campaign_name" -> event.properties.campaign_name).asJson,
+        "time" -> Json.fromString(event.time),
+    )
+  }
+}
+
+object BrazeSubscribeEvent {
+
+  def apply(externalId: String, sub: EmailNewsletter, isSubscribed: Boolean, timestamp: Instant, updateExistingOnlyField: Boolean = false): List[BrazeEvent] = {
+    // Marketing need to be able to segment subscription events by campaign. To do this the campaign name must be in the name of the event,
+    // (as braze can only segment by custom event name not property).
+    val newsletterEventName = if (isSubscribed) s"${sub.brazeSubscribeEventNamePrefix}_subscribe_email_date" else s"${sub.brazeSubscribeEventNamePrefix}_unsubscribe_email_date"
+
+    val generalEventName = if (isSubscribed) "EditorialSubscribe" else "EditorialUnsubscribe"
+    // TODO: Do we need to handle subscription? Or just unsubscribe?
```

Comment:
I think it's worth changing this to only allow unsubscribe events, but with the code having an element of redundancy (e.g. having the `generalEventName` `val` still present) in case we ever want to make an easy swap back.